src/test/kotlin/three/consulting/epoc/service/ProjectServiceIntegrationTest.kt

```diff
@@ -116,6 +119,89 @@ class ProjectServiceIntegrationTest : IntegrationTest() {
+        val invalidProject = ProjectDTO(
+            name = "End date precedes start date",
+            description = "Project fail",
+            startDate = LocalDate.parse("2021-11-11"),
+            endDate = LocalDate.parse("2011-11-11"),
+            customer = CustomerDTO(1, "Failing Project Customer"),
+            managingEmployee = EmployeeDTO(1, "Failure", "Project-worker", "new.project@worker.fi"),
+        )
+        Assertions.assertThatThrownBy { projectService.createProject(invalidProject) }
+            .isInstanceOf(UnableToCreateProjectException::class.java)
+            .hasMessage("Cannot create a project with end date preceding start date.")
+    }
+
+    @Test
+    fun `create project with null end date`() {
+        val project = ProjectDTO(
+            name = "Sample",
+            description = "Sample project with null end date",
+            startDate = LocalDate.now(),
+            endDate = null,
+            customer = CustomerDTO(1, "New Project Customer"),
+            managingEmployee = EmployeeDTO(1, "New", "Project-worker", "new.project@worker.fi"),
+            status = Status.ARCHIVED,
+        )
+        val addedProject: ProjectDTO = projectService.createProject(project)
+        assertThat(addedProject.name).isEqualTo(project.name)
+        assertThat(addedProject.description).isEqualTo(project.description)
+        assertThat(addedProject.startDate).isEqualTo(project.startDate)
+        assertThat(addedProject.endDate).isEqualTo(project.endDate)
+        assertThat(addedProject.status).isEqualTo(project.status)
+    }
+
+    @Test
+    fun `update project with end date preceding start date raises error`() {
+        val invalidProject = ProjectDTO(
+            id = 1,
+            name = "asd",
+            customer = CustomerDTO(1, "Updating Project customer"),
+            managingEmployee = EmployeeDTO(1, "Updating", "Project-Worker", "updateingProject@worker.fi"),
+            startDate = LocalDate.now(),
+            endDate = LocalDate.now().minusDays(1),
+        )
+        Assertions.assertThatThrownBy { projectService.updateProjectForId(invalidProject) }
+            .isInstanceOf(UnableToUpdateProjectException::class.java)
+            .hasMessage("Cannot update a project with end date preceding start date.")
+    }
+
+    @Test
+    fun `update project with null end date passes`() {
+        val existingProject = projectService.findProjectForId(1)
+        val project = ProjectDTO(
+            id = 1,
+            name = "asd",
+            customer = CustomerDTO(1, "Updating Project customer"),
+            managingEmployee = EmployeeDTO(1, "Updating", "Project-Worker", "updateingProject@worker.fi"),
+            startDate = LocalDate.now(),
+            endDate = null,
+        )
+        val updatedProject = projectService.updateProjectForId(project)
+        if (existingProject != null) {
+            assertThat(updatedProject.name).isNotEqualTo(existingProject.name)
+        }
```

Comment:
Avoid using if/else statements in tests. Now here you might not end up in the assertion, which is not desired. You can use 
```kotlin
assertThat(updatedProject.name).isNotEqualTo(existingProject?.name)
```
Then the assertion will fail if existingProject is null, because you're comparing existing value to `null`